packages/sdk-core/src/SuperToken.ts

```diff
@@ -141,6 +145,9 @@ export default abstract class SuperToken extends ERC20Token {
             const nativeTokenSymbol = resolverData.nativeTokenSymbol || "ETH";
             const nativeSuperTokenSymbol = nativeTokenSymbol + "x";
 
+            const tokenSymbol = await superToken
+                .connect(options.provider)
+                .symbol();
             if (nativeSuperTokenSymbol === tokenSymbol) {
```

Comment:
Not a fan of this condition for checking whether it's a `NativeAssetSuperToken` -- I wish it was more concrete than just checking the token symbol and making assumptions.

Reply:
the only other check I can think of is checking if a function unique to `NativeAssetSuperToken`'s exist: e.g. `upgradeByETH` - it requires casting the superToken contract as any and I'm not sure if it actually works.